crates/oxc_sourcemap/src/encode.rs

```diff
@@ -14,20 +15,35 @@ pub fn encode(sourcemap: &SourceMap) -> String {
         buf.push_str("\",");
     }
     buf.push_str("\"names\":[");
-    buf.push_str(&sourcemap.names.iter().map(|x| format!("{x:?}")).collect::<Vec<_>>().join(","));
+    let names = sourcemap
+        .names
+        .iter()
+        .map(|x| serde_json::to_string(x.as_ref()))
+        .collect::<std::result::Result<Vec<_>, serde_json::Error>>()
+        .map_err(Error::from)?;
+    buf.push_str(&names.join(","));
     buf.push_str("],\"sources\":[");
-    buf.push_str(&sourcemap.sources.iter().map(|x| format!("{x:?}")).collect::<Vec<_>>().join(","));
+    let sources = sourcemap
+        .sources
+        .iter()
+        .map(|x| serde_json::to_string(x.as_ref()))
```

Comment:
I think we should document why we need `serde_json::to_string`, and we should have a test case for this at least.

Reply:
Done